apps/builder/app/builder/features/style-panel/style-source-section.tsx

```diff
@@ -341,53 +352,118 @@ export const StyleSourcesSection = () => {
-      onCreateItem={createStyleSource}
-      onSelectAutocompleteItem={({ id }) => {
-        addStyleSourceToInstace(id);
-      }}
-      onDuplicateItem={(id) => {
-        const newId = duplicateStyleSource(id);
-        if (newId !== undefined) {
-          setEditingItemId(newId);
-        }
-      }}
-      onConvertToToken={(id) => {
-        convertLocalStyleSourceToToken(id);
-        setEditingItemId(id);
-      }}
-      onRemoveItem={(id) => {
-        removeStyleSourceFromInstance(id);
-      }}
-      onDeleteItem={(id) => {
-        deleteStyleSource(id);
-      }}
-      onSort={(items) => {
-        reorderStyleSources(items.map((item) => item.id));
-      }}
-      onSelectItem={(styleSourceSelector) => {
-        createLocalStyleSourceIfNotExists(styleSourceSelector.styleSourceId);
-        selectedStyleSourceSelectorStore.set(styleSourceSelector);
-      }}
-      // style source renaming
-      editingItemId={editingItemId}
-      onEditItem={(id) => {
-        setEditingItemId(id);
-        // prevent deselect after renaming
-        if (id !== undefined) {
-          selectedStyleSourceSelectorStore.set({
-            styleSourceId: id,
-          });
+    <>
+      <StyleSourceInput
+        items={items}
+        value={value}
+        selectedItemSelector={selectedOrLastStyleSourceSelector}
+        componentStates={componentStates}
+        onCreateItem={createStyleSource}
+        onSelectAutocompleteItem={({ id }) => {
+          addStyleSourceToInstace(id);
+        }}
+        onDuplicateItem={(id) => {
+          const newId = duplicateStyleSource(id);
+          if (newId !== undefined) {
+            setEditingItemId(newId);
+          }
+        }}
+        onConvertToToken={(id) => {
+          convertLocalStyleSourceToToken(id);
+          setEditingItemId(id);
+        }}
+        onRemoveItem={(id) => {
+          removeStyleSourceFromInstance(id);
+        }}
+        onDeleteItem={(id) => {
+          const token = availableStyleSources.find(
+            (source) => source.id === id
+          );
+          if (token?.type === "token") {
+            setTokenToDelete(token);
+          }
+        }}
+        onSort={(items) => {
+          reorderStyleSources(items.map((item) => item.id));
+        }}
+        onSelectItem={(styleSourceSelector) => {
+          createLocalStyleSourceIfNotExists(styleSourceSelector.styleSourceId);
+          selectedStyleSourceSelectorStore.set(styleSourceSelector);
+        }}
+        // style source renaming
+        editingItemId={editingItemId}
+        onEditItem={(id) => {
+          setEditingItemId(id);
+          // prevent deselect after renaming
+          if (id !== undefined) {
+            selectedStyleSourceSelectorStore.set({
+              styleSourceId: id,
+            });
+          }
+        }}
+        onChangeItem={(item) => {
+          renameStyleSource(item.id, item.label);
+        }}
+      />
+      {tokenToDelete && (
```

Comment:
Why not use open prop?

Reply:
dunno, felt semantically like I only want to render the component when we are in the right state, what's the downside?

Reply:
Inconsistent usage and lack of transition when there is any.

Reply:
done